engine/src/main/java/org/terasology/rendering/nui/editor/utils/NUIEditorMenuTreeBuilder.java

```diff
@@ -250,7 +252,7 @@ private void populateContextMenu(JsonTree node, MenuTree addTree, boolean isSkin
                 // If the node is part of a layout, add an option to add the layoutInfo node.
                 if (nodeInfo.getLayoutClass() != null) {
                     String layoutInfo = "layoutInfo";
-                    if (!node.hasChildWithKey(layoutInfo)) {
+                    if (!Objects.equals(node.getValue().getKey(), layoutInfo) && !node.hasChildWithKey(layoutInfo)) {
```

Comment:
Since you already have a way to calculate the possible child nodes for layoutInfo. Maybe the criteria for adding a layoutInfo node could be that it is possible to add child nodes to it. So that it is not possibe to add layoutInfo nodes that can't have any child.

Otherwise based on UIWidgetTypeAdapter the logic I think the condition for adding a layoutInfo node should be:
* The node represents a UIWidget
* The parent is `UIFormat.CONTENTS_FIELD`
* The parent of the parent implements UILayout
* The hint type of the layout can be instanced (e.g. no layoutInfo for CardLayout)

I added comments to the following code part of `UIWidgetTypeAdapter` and added comments what it means for our condition:
```
if (UILayout.class.isAssignableFrom(elementMetadata.getType())) { // => parent of parent must implement UILayout
                UILayout<LayoutHint> layout = (UILayout<LayoutHint>) element;

                Class<? extends LayoutHint> layoutHintType = (Class<? extends LayoutHint>)
                    ReflectionUtil.getTypeParameter(elementMetadata.getType().getGenericSuperclass(), 0);
                if (jsonObject.has(CONTENTS_FIELD)) { // parent => is CONTENTS_FIELD
                    for (JsonElement child : jsonObject.getAsJsonArray(CONTENTS_FIELD)) {
                        UIWidget childElement = context.deserialize(child, UIWidget.class); // => element is UIWidget
                        if (childElement != null) {
                            LayoutHint hint = null;
                            if (child.isJsonObject()) {
                                JsonObject childObject = child.getAsJsonObject();
                                if (layoutHintType != null && !layoutHintType.isInterface() && !Modifier.isAbstract(layoutHintType.getModifiers()) // => layoutHintType can be instanced (no interface no abstract class)
                                    && childObject.has(LAYOUT_INFO_FIELD)) { 
                                    hint = context.deserialize(childObject.get(LAYOUT_INFO_FIELD), layoutHintType);
                                }
                            }
                            layout.addWidget(childElement, hint);
                        }
                    }
                }
            }
```